third_party/terraform/utils/import.go

```diff
@@ -25,7 +26,24 @@ func parseImportId(idRegexes []string, d TerraformResourceData, config *Config)
 			// Starting at index 1, the first match is the full string.
 			for i := 1; i < len(fieldValues); i++ {
 				fieldName := re.SubexpNames()[i]
-				d.Set(fieldName, fieldValues[i])
+				// Here we have a challenge.  This will work properly in production, but
+				// in testing, we have a problem.  In test mode, instead of returning an
+				// error, a failed d.Set() call will panic.  In production, we catch the
+				// error and retry, but in testing, if we ever have a component of an ID
+				// which is strictly base-10 numeric, we're going to crash with a weird
+				// panic.  At least the panic will be isolated here, so you'll be able to
+				// find this comment.  :)  I promise the error that brought you here is
+				// not one that will show up in production!  Try to change the ID so that
+				// it doesn't contain a strictly numeric component.
+				if atoi, atoiErr := strconv.Atoi(fieldValues[i]); atoiErr == nil {
```

Comment:
Wait - why do we need to do all this exactly? If a user creates a rule with priority `100` couldn't they they just type `terraform import ....../100` to import it? This is unclear and the comment overhead only confuses me more.

Reply:
Thanks for the explanation! Unfortunately, I have sad news. It will already cause a panic :( https://github.com/terraform-providers/terraform-provider-google-beta/blob/2.0.0/website/docs/r/accesscontextmanager_access_policy.html.markdown#import

Do you have any other ideas off the top of your head for solving this? Looking briefly, have we tried `d.SetType` or `d.UnsafeSetFieldRaw`? Another possibly more adventurous idea: if we do a `d.Get`, will the nil/none value returned be appropriately typed?

Reply:
Ooh, nice sneaky plan with `d.Get`.  The docs say it oughtta work - https://github.com/hashicorp/terraform/blob/master/helper/schema/resource_data.go#L64.  I'll give that a try!

Reply:
Tests pass.  Nice!